src/globus_cli/login_manager.py

```diff
@@ -0,0 +1,87 @@
+import functools
+
+import click
+
+from .tokenstore import token_storage_adapter
+
+
+class LoginManager:
+    def __init__(self):
+        self._token_storage = token_storage_adapter()
+
+    def has_login(self, resource_server: str):
+        """
+        Determines if the user has a refresh token for the given
+        resource server
+        """
+        tokens = self._token_storage.get_token_data(resource_server)
+        return tokens is not None and "refresh_token" in tokens
+
+
+def requires_login(*args: str, pass_manager: bool = False):
+    """
+    Command decorator for specifying a resource server that the user must have
+    tokens for in order to run the command.
+
+    Simple usage for commands that have static resource needs: simply list all
+    needed resource servers as args:
+
+    @requries_login("auth.globus.org")
+
+    @requires_login("auth.globus.org", "transfer.api.globus.org")
+
+    Usage for commands which have dynamic resource servers depending
+    on the arguments passed to the command (e.g. commands for the GCS API)
+
+    @requies_login(pass_manager=True)
+    def command(login_manager, endpoint_id)
+
+        login_manager.<do the thing>(endpoint_id)
+
+    """
+    resource_servers = args
+
+    def inner(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
```

Comment:
Question: should this whole wrapper function be functionality of the LoginManager itself? i.e. `LoginManager.check_required_logins(*args)` with a body nearly identical to this function.

My thought is that in the cases for `pass_manager=True`, we will want to presumably trigger this same functionality off of the `LoginManager`.

We can save any potential refactor for later if you prefer, however.